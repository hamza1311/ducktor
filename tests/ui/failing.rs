use ducktor::FromJsValue;

#[derive(FromJsValue)]
enum Data {}

#[derive(FromJsValue)]
union Union {

}

#[derive(FromJsValue)]
struct UnitStruct;

#[derive(FromJsValue)]
struct TupleStruct(u32, i64);

fn main() {}