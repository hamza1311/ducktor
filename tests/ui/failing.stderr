error: cannot derive `FromJsValue` for enums
 --> tests/ui/failing.rs:4:1
  |
4 | enum Data {}
  | ^^^^

error: cannot derive `FromJsValue` for unions
 --> tests/ui/failing.rs:7:1
  |
7 | union Union {
  | ^^^^^

error: cannot derive `FromJsValue` for unit structs
  --> tests/ui/failing.rs:12:1
   |
12 | struct UnitStruct;
   | ^^^^^^^^^^^^^^^^^^

error: cannot derive `FromJsValue` for tuple structs
  --> tests/ui/failing.rs:15:19
   |
15 | struct TupleStruct(u32, i64);
   |                   ^^^^^^^^^^

error: unions cannot have zero fields
 --> tests/ui/failing.rs:7:1
  |
7 | / union Union {
8 | |
9 | | }
  | |_^
